core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -147,6 +168,18 @@ private synchronized void init(MasterInquireClient masterInquireClient) {
         new FileSystemMasterClientPool(mParentSubject, mMasterInquireClient);
     mBlockMasterClientPool = new BlockMasterClientPool(mParentSubject, mMasterInquireClient);
     mClosed.set(false);
+    // setup metrics master client sync
+    mMetricsMasterClient = new MetricsMasterClient(MasterClientConfig.defaults());
```

Comment:
Here, I think you will have to propagate the master inquire client as well as the subject.

`MasterClientConfig.defaults().withSubject(mParentSubject).withMasterInquireClient(mMasterInquireClient)`

Reply:
That's good to know. Thanks!